src/dex/bebop/bebop.ts

```diff
@@ -0,0 +1,744 @@
+          instruction.targetQuote
+            ? remaining.div(price)
+            : remaining.times(price),
+        );
+        filled = true;
+        break;
+      }
+    }
+    if (filled) {
+      return output;
+    } else {
+      return BigNumber(0);
+    }
+  }
+
+  calculateOutput(
+    instructions: RoutingInstruction[],
+    srcToken: Token,
+    destToken: Token,
+    amounts: bigint[],
+    side: SwapSide,
+  ): bigint[] {
+    const outputs = [];
+    const inputDecimals =
+      side == SwapSide.SELL ? srcToken.decimals : destToken.decimals;
+    const outputDecimals =
+      side == SwapSide.SELL ? destToken.decimals : srcToken.decimals;
+
+    for (const amount of amounts) {
+      if (amount == 0n) {
+        outputs.push(0n);
+        continue;
+      }
+      const amountDecimals = BigNumber(amount.toString()).div(
+        getBigNumberPow(inputDecimals),
+      );
+      let output: BigNumber = BigNumber(0);
+      for (const instruction of instructions) {
+        output = this.runInstruction(
+          instruction,
+          output.gt(0) ? output : amountDecimals,
+        );
+        if (output.eq(0)) {
+          break;
+        }
+      }
+      if (output.gt(0)) {
+        outputs.push(
+          BigInt(output.times(getBigNumberPow(outputDecimals)).toFixed(0)),
+        );
+      } else {
+        outputs.push(0n);
+      }
+    }
+    return outputs;
+  }
+
+  // Returns pool prices for amounts.
+  // If limitPools is defined only pools in limitPools
+  // should be used. If limitPools is undefined then
+  // any pools can be used.
+  async getPricesVolume(
+    srcToken: Token,
+    destToken: Token,
+    amounts: bigint[],
+    side: SwapSide,
+    blockNumber: number,
+    limitPools?: string[],
+  ): Promise<null | ExchangePrices<BebopData>> {
+    this.tokensMap = (await this.getCachedTokens()) || {};
+
+    try {
+      const pools =
+        limitPools ??
+        (await this.getPoolIdentifiers(srcToken, destToken, side, blockNumber));
+
+      if (pools.length === 0) {
+        return null;
+      }
+
+      const instructions = await this.calculateInstructions(
+        srcToken,
+        destToken,
+        side,
+      );
+
+      if (!instructions) {
+        return null;
+      }
+
+      const outputs = this.calculateOutput(
+        instructions,
+        srcToken,
+        destToken,
+        amounts,
+        side,
+      );
+
+      // Up to 3bps deviation is expected in the output compared to the on-chain result
+      // On SwapSide.Sell, outputs compared  to quoting are coming out: -0.1 bips -> USDC, -1 bips Alt -> Alt.
+      // On SwapSide.Buy, outputs compared to quoteing are coming out: 0.1 bips -> USDC, 1-3 bips Alt -> Alt.
+
+      const outDecimals = SwapSide.SELL
+        ? destToken.decimals
+        : srcToken.decimals;
+
+      return [
+        {
+          prices: outputs,
+          unit: BigInt(outDecimals),
+          data: {},
+          poolIdentifier: pools[0],
```

Comment:
maybe better `instructions.pair` ? cause looks like pools are not used in the pricing

Reply:
I'd rather leave what was calculated by `getPoolIdentifiers`